test/e2e/proxy_test.go

```diff
@@ -68,7 +70,400 @@ func skipIfProxyDisabled(t *testing.T) {
+	busybox1 := "busybox-1"
+	node0 := nodeName(0)
+	node1 := nodeName(1)
+	clusterIngressIP := []string{"169.254.169.1"}
+	localIngressIP := []string{"169.254.169.2"}
+	busybox0Ipv4, busybox0Ipv6 := createTestClientPod(t, data, busybox0, node0)
+	busybox1Ipv4, busybox1Ipv6 := createTestClientPod(t, data, busybox1, node1)
+	busybox0IP := busybox0Ipv4
+	busybox1IP := busybox1Ipv4
+	if isIPv6 {
+		clusterIngressIP = []string{"fd75::aabb:ccdd:ef00"}
+		localIngressIP = []string{"fd75::aabb:ccdd:ef01"}
+		busybox0IP = busybox0Ipv6
+		busybox1IP = busybox1Ipv6
+	}
+	createLoadBalancerService(t, data, "agnhost-cluster", clusterIngressIP, false, isIPv6)
+	createLoadBalancerService(t, data, "agnhost-local", localIngressIP, true, isIPv6)
+	port := "8080"
+	clusterUrl := net.JoinHostPort(clusterIngressIP[0], port)
+	localUrl := net.JoinHostPort(localIngressIP[0], port)
+
+	createAgnhostPod(t, data, agnhost0, node0, false)
+	createAgnhostPod(t, data, agnhost1, node1, false)
+	t.Run("Pod CIDR Endpoints", func(t *testing.T) {
+		loadBalancerTestCases(t, data, clusterUrl, localUrl, busybox0, busybox1, busybox0IP, busybox1IP, agnhost0, agnhost1)
+	})
+	require.NoError(t, data.deletePod(testNamespace, agnhost0))
+	require.NoError(t, data.deletePod(testNamespace, agnhost1))
+
+	createAgnhostPod(t, data, hostAgnhost0, node0, true)
+	createAgnhostPod(t, data, hostAnghost1, node1, true)
+	t.Run("Host Network Endpoints", func(t *testing.T) {
+		loadBalancerTestCases(t, data, clusterUrl, localUrl, busybox0, busybox1, busybox0IP, busybox1IP, node0, node1)
+	})
+}
+
+func loadBalancerTestCases(t *testing.T, data *TestData, clusterUrl, localUrl, busybox0, busybox1, busybox0IP, busybox1IP, hostname0, hostname1 string) {
+	t.Run("ExternalTrafficPolicy:Cluster/Client:Node", func(t *testing.T) {
+		testLoadBalancerClusterFromNode(t, data, clusterUrl)
+	})
+	t.Run("ExternalTrafficPolicy:Cluster/Client:Pod", func(t *testing.T) {
+		testLoadBalancerClusterFromPod(t, data, clusterUrl, busybox0, busybox1)
+	})
+	t.Run("ExternalTrafficPolicy:Local/Client:Node", func(t *testing.T) {
+		testLoadBalancerLocalFromNode(t, data, localUrl, hostname0, hostname1)
+	})
+	t.Run("ExternalTrafficPolicy:Local/Client:Pod", func(t *testing.T) {
+		testLoadBalancerLocalFromPod(t, data, localUrl, busybox0, busybox1, busybox0IP, busybox1IP, hostname0, hostname1)
+	})
+}
+
+func createLoadBalancerService(t *testing.T, data *TestData, serviceName string, ingressIPs []string, nodeLoadExternal, isIPv6 bool) {
+	ipProctol := corev1.IPv4Protocol
+	if isIPv6 {
+		ipProctol = corev1.IPv6Protocol
+	}
+	_, err := data.createAgnhostLoadBalancerService(serviceName, true, nodeLoadExternal, ingressIPs, &ipProctol)
+	require.NoError(t, err)
+}
+
+func testLoadBalancerClusterFromNode(t *testing.T, data *TestData, url string) {
+	skipIfKubeProxyEnabled(t, data)
+	errMsg := "Server LoadBalancer whose externalTrafficPolicy is Cluster should be able to be connected from Node"
+	probeFromNode(t, nodeName(0), url, errMsg)
```

Comment:
I feel passing error message to a function and make it log it when failed is not a common pattern.
Instead, the function should just return error and let the caller to log the error or ignore it. 
```
require.NoError(t, probeFromNode(t, nodeName(0), url), "Server LoadBalancer whose externalTrafficPolicy is Cluster should be able to be connected from Node")
```